bindata/network/multus/multus.yaml

```diff
@@ -455,3 +455,142 @@ spec:
+                  fi
+                  if [ -z ${KUBERNETES_SERVICE_PORT} ]; then
+                    error "KUBERNETES_SERVICE_PORT not set"; exit 1;
+                  fi
+
+                  if [ "$SKIP_TLS_VERIFY" == "true" ]; then
+                    TLS_CFG="insecure-skip-tls-verify: true"
+                  elif [ -f "$KUBE_CA_FILE" ]; then
+                    TLS_CFG="certificate-authority-data: $(cat $KUBE_CA_FILE | base64 | tr -d '\n')"
+                  fi
+
+                  # Write a kubeconfig file for the CNI plugin.  Do this
+                  # to skip TLS verification for now.  We should eventually support
+                  # writing more complete kubeconfig files. This is only used
+                  # if the provided CNI network config references it.
+                  touch $WHEREABOUTS_KUBECONFIG
+                  chmod ${KUBECONFIG_MODE:-600} $WHEREABOUTS_KUBECONFIG
+                  cat > $WHEREABOUTS_KUBECONFIG <<EOF
+                # Kubeconfig file for Multus CNI plugin.
+                apiVersion: v1
+                kind: Config
+                clusters:
+                - name: local
+                  cluster:
+                    server: ${KUBERNETES_SERVICE_PROTOCOL:-https}://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}
+                    $TLS_CFG
+                users:
+                - name: whereabouts
+                  user:
+                    token: "${SERVICEACCOUNT_TOKEN}"
+                contexts:
+                - name: whereabouts-context
+                  context:
+                    cluster: local
+                    user: whereabouts
+                    namespace: ${WHEREABOUTS_NAMESPACE}
+                current-context: whereabouts-context
+                EOF
+
+                # Kubeconfig file for Multus CNI plugin.
+                cat > $WHEREABOUTS_GLOBALCONFIG <<EOF
+                {
+                  "datastore": "kubernetes",
+                  "kubernetes": {
+                    "kubeconfig": "/etc/kubernetes/cni/net.d/whereabouts.d/whereabouts.kubeconfig"
+                  },
+                  "log_level": "debug"
+                }
+                EOF
+
+                else
+                  warn "Doesn't look like we're running in a kubernetes environment (no serviceaccount token)"
+                fi
+
+                # copy whereabouts to the cni bin dir
+                # SKIPPED DUE TO FIPS COPY.
+                # cp -f /whereabouts $CNI_BIN_DIR
+
+                # ---------------------- end Generate a "kube-config".
+
+                # Unless told otherwise, sleep forever.
+                # This prevents Kubernetes from restarting the pod repeatedly.
+                should_sleep=${SLEEP:-"true"}
+                echo "Done configuring CNI.  Sleep=$should_sleep"
+                while [ "$should_sleep" == "true"  ]; do
+                    sleep 1000000000000
+                done
```

Comment:
Ah! I found it available on the correct path, which is actually referenced by the `SystemCNIConfDir` variable:
https://github.com/openshift/cluster-network-operator/blob/87d4207091c6a6aed36ca9cf6f190c79c9a8d456/pkg/network/multus.go#L14

But, the file is available on host with r/w permissions only to the root user:
```
sh-4.4# ls -la /etc/kubernetes/cni/net.d/whereabouts.d/
total 16
drwxr-xr-x. 2 root root    60 Oct 14 09:14 .
drwxr-xr-x. 4 root root    65 Oct 14 09:30 ..
-rw-r--r--. 1 root root   162 Oct 14 09:29 whereabouts.conf
-rw-------. 1 root root 11559 Oct 14 09:29 whereabouts.kubeconfig

```
Which causes our humble pod to not be able to access it:
```
oc logs -nopenshift-multus ip-reconciler-27236755--1-zzwlw
2021-10-14T09:55:44Z [debug] NewReconcileLooper - Kubernetes config file located at: /host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig
2021-10-14T09:55:44Z [error] failed to instantiate the Kubernetes client: Error loading config file "/host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig": open /host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig: permission denied
2021-10-14T09:55:44Z [error] failed to create the reconcile looper: failed to instantiate the Kubernetes client: Error loading config file "/host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig": open /host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig: permission denied

```

Reply:
Hrmmm, I'm a little torn. So, all the other CNI components herein are privileged. So it's not... weird. 

Maybe it's even appropriate. Because if we make that file more easily readable, then, that service account credentials are exposed.